packages/valory/skills/transaction_settlement_abci/rounds.py

```diff
@@ -155,16 +154,13 @@ def to_be_validated_tx_hash(self) -> str:
         due to the way we are inserting the hashes in the array.
         We keep the hashes sorted by the time of their finalization.
         If this property is accessed before the finalization succeeds,
-        then it is incorrectly used and raises an internal error.
+        then it is incorrectly used and raises an error.
 
         :return: the tx hash which is ready for validation.
         """
-        if len(self.tx_hashes_history) > 0:
-            return self.tx_hashes_history[-1]
-        raise ABCIAppInternalError(  # pragma: no cover
-            "An Error occurred while trying to get the tx hash for validation: "
-            "There are no transaction hashes recorded!"
-        )
+        if not self.tx_hashes_history:
+            raise ValueError("FSM design error: tx hash should exist")
+        return self.tx_hashes_history[-1]
```

Comment:
Why do we still raise here? Is it possible to avoid this by updating `tx_hashes_history` to use `get_strict`?

Reply:
we could do that. But what would be the benefit? We would also end up raising a `ValueError`